pulp_ansible/tests/functional/api/collection/v3/test_collection_naming_edgecases.py

```diff
@@ -0,0 +1,336 @@
+    # api._handle_202(client._cfg, response, client.pulp_host)
+    if response.request.method == "POST":
+        # import epdb; epdb.st()
+        task_url = response.json()["task"]
+        task = client.get(task_url)
+        while task.get("finished_at") is None:
+            # print(task['state'])
+            task = client.get(task_url)
+        # print(task['state'])
+        return task
+    else:
+        return response.json()
+
+
+def get_galaxy_url(base, path):
+    cfg = config.get_config()
+    path = path.lstrip("/")
+    GALAXY_API_ROOT = cfg.custom.get(
+        "galaxy_api_root", "/pulp_ansible/galaxy/%(base_path)s/api/"
+    ) % {"base_path": base}
+    return urljoin(GALAXY_API_ROOT, path)
+
+
+def get_collection_versions_by_repo(pulp_client, repo_names=None):
+    """Make a mapping of repositories and their CV content."""
+
+    repos = get_repositories(pulp_client)
+    dists = get_distributions(pulp_client)
+
+    if repo_names is None:
+        repos = get_repositories(pulp_client)
+        repo_names = list(repos.keys())
+
+    cvs = {}
+
+    for repo_name in repo_names:
+        next_url = (
+            "/pulp_ansible/galaxy/"
+            + repo_name
+            + "/api/v3/plugin/ansible/content/"
+            + repo_name
+            + "/collections/index/"
+        )
+        while next_url:
+            resp = pulp_client.get(next_url)
+            next_url = resp["links"]["next"]
+            for collection in resp["data"]:
+                next_cv_url = collection["versions_url"]
+                while next_cv_url:
+                    cvresp = pulp_client.get(next_cv_url)
+                    next_cv_url = cvresp["links"]["next"]
+                    for cv_summary in cvresp["data"]:
+                        cv_info = pulp_client.get(cv_summary["href"])
+
+                        # include the repoid & distid ...
+                        cv_info["repository_id"] = repos[repo_name]["pulp_href"].split("/")[-2]
+                        if repo_name in dists:
+                            did = dists[repo_name]["pulp_href"]
+                            did = did.split("/")[-2]
+                            cv_info["distribution_id"] = did
+                            cv_info["base_path"] = dists[repo_name]["base_path"]
+                        else:
+                            cv_info["distribution_id"] = None
+                            cv_info["base_path"] = None
+
+                        key = (
+                            repo_name,
+                            cv_info["namespace"]["name"],
+                            cv_info["name"],
+                            cv_info["version"],
+                        )
+                        cvs[key] = cv_info
+
+    return cvs
+
+
+def get_collection_versions(pulp_client):
+    cversions = pulp_client.get("/pulp/api/v3/content/ansible/collection_versions/")
+    return dict(((x["namespace"], x["name"], x["version"]), x) for x in cversions)
```

Comment:
We are moving towards using fixtures that encapsulate bindings classes.
This one should do the trick:
https://github.com/pulp/pulp_ansible/blob/main/pulp_ansible/tests/functional/conftest.py#L91